persistence-api/src/main/java/io/stargate/db/schema/Column.java

```diff
@@ -163,21 +183,12 @@ default Object validate(Object value, String location) throws ValidationExceptio
     @Value.Lazy
     TypeCodec codec();
```

Comment:
Not specific to this PR, but I think we should consider dropping the dependency to the driver. It's currently used for:
* the type codec infrastructure
* `UdtValue` and `TupleValue`
* `GettableByIndex` and `GettableByName` (implemented by `Row`)
* a few types from its transitive dependencies: nullability annotations, SLF4J

I understand that it's convenient to reuse some of the code, but from a logical point of view it's not justified, the persistence API does not "use" the driver.

Also the integration is not perfect, the driver API relies heavily on the driver's types, but the persistence API has its own type representation. This leaks in a few places, for example `ArrayListBackedRow` needs to implement `getType` which returns a driver type.

Reply:
I agree, and that's my long term goal as well. Just not something for today :)